include/picongpu/fields/MaxwellSolver/YeePML/YeePML.hpp

```diff
@@ -362,11 +399,21 @@ namespace picongpu
                  */
                 void update_beforeCurrent(uint32_t const currentStep)
                 {
-                    /* These steps are the same as in the Yee solver,
-                     * PML updates are done as part of solver.updateE( ),
-                     * solver.updateBHalf( )
+                    /* These steps are the same as in the Yee solver, PML updates are done as part of methods of
+                     * solver. Note that here we do the second half of updating B, thus completing the first half
+                     * started in a call to update_afterCurrent() at the previous time step. This splitting of B update
+                     * is standard for Yee-type field solvers in PIC codes due to particle pushers normally requiring E
+                     * and B values defined at the same time while the field solver operates with time-staggered
+                     * fields. However, while the standard Yee solver in vacuum is linear in a way of two consecutive
+                     * updates by dt/2 being equal to one update by dt, this is not true for the convolutional field
+                     * updates in PML. Thus, for PML we have to distinguish between the updates by dt/2 by introducing
+                     * first and second halves of the update. This distinction only concerns the convolutional field B
+                     * data used inside the PML, and not the full fields used by the rest of the code. In the very
+                     * first time step of a simulation we start with the second half right away, but this is no
+                     * problem, since the only meaningful initial conditions in the PML area are zero for the
+                     * to-be-absorbed components.
                      */
-                    solver.template updateBHalf<CORE + BORDER>(currentStep);
+                    solver.template updateBSecondHalf<CORE + BORDER>(currentStep);
```

Comment:
Why do you not call `updateBHalf`, since this function is part of YeePML there should be no need to hide that there is a bool parameter as second argument.

Reply:
I thought to do it, but decided to express more meaning of this parameter via additional function names. Since this is what it means: are we doing the first or second half

Reply:
So basically first and second half are terms on the level of the caller here. While the updating of convolutional components is internal terms that I did not want to expose here.